lib/carrierwave/uploader/versions.rb

```diff
@@ -325,11 +325,11 @@ def remove_versions!
       end
 
       def retrieve_versions_from_cache!(cache_name)
-        active_versions.each { |name, v| v.retrieve_from_cache!(cache_name) }
+        versions.each { |name, v| v.retrieve_from_cache!(cache_name) }
       end
 
       def retrieve_versions_from_store!(identifier)
-        active_versions.each { |name, v| v.retrieve_from_store!(identifier) }
+        versions.each { |name, v| v.retrieve_from_store!(identifier) }
       end
 
```

Comment:
We don't need to evaluate the condition here, like before the changes in https://github.com/carrierwaveuploader/carrierwave/pull/1351 .
We can keep all versions retrieved internally, and only evaluate them when users accesses.